GSA-TTS/FAC
# ESLint Reconfiguration and Related Fixes

# ESLint Reconfiguration and Related Fixes

Issue: https://github.com/GSA-TTS/FAC/issues/3933

## Changes:
1. Have a standard `npx eslint` cover all the files we need covered
    * Update `eslint.config.mjs` to handle standard options, such as ignored directories
    * Update the package scripts and Makefile to use the standard execution
    * Now covers Cypress, it's own config file, etc.
2. Migrate a number of files to ES modules format
    * ES modules have been supported by node since 2017, so we're good there
    * Cypress config and plugin index files are updates, as well as the ESBuild specific `build-assets.mjs`
    * _The future is now_
3. Fix all the ESLint errors. 
    * Most are a Cypress error akin to `It is unsafe to chain further commands that rely on the subject after this command.` This means we shouldn't clear a text box, type into it, and blur it in rapid succession. These just get broken into several statements where re recapture the text box each time to verify that it's okay. 
    * A bunch of ES module changes
    * That's pretty much it? We already used prettier for just about everything, so there's more actual errors/warnings than formatting errors. 

Also, I included the ESLint Babel parser (@babel/eslint-parser). ESLint only officially supports the latest final ECMAScript standard, so Babel helps keep everything non-ECMA in check.

## How to test:
Locally:
1. Switch to this branch and run normally.
2. Bring up a fresh stack, including refreshing the `node_modules` directory and running a `collectstatic`
3. Run `npx eslint` to verify the new config works.
5. Run the Cypress tests, to verify that the not-insignificant changes didn't change the behavior.

Other things:
1. The GitHub Actions PR tests should still pass.
2. Cypress through GitHub Actions should also still pass.


## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
